packages/headless/src/features/commerce/parameters/parameters-serializer.ts

```diff
@@ -36,12 +58,108 @@ const supportedFacetParameters: Record<FacetParameters, boolean> = {
   df: true,
 };
 
-// eslint-disable-next-line @cspell/spellchecker
-// TODO CAPI-907: Handle sort and pagination
 function serialize(parameters: CommerceSearchParameters): string {
   return coreSerialize(serializePair)(parameters);
 }
 
+function serializePair(pair: [string, unknown]) {
+  const [key, val] = pair;
+
+  if (!isValidKey(key)) {
+    return '';
+  }
+
+  if (key === 'sortCriteria') {
+    return isSortCriteriaObject(val) ? serializeSortCriteria(key, val) : '';
+  }
+
+  if (keyHasObjectValue(key) && !isRangeFacetKey(key)) {
+    return isFacetObject(val) ? serializeFacets(key, val) : '';
+  }
+
+  if (key === 'nf' || key === 'df') {
+    return isRangeFacetObject(val) ? serializeRangeFacets(key, val) : '';
+  }
+
+  return serializeSpecialCharacters(key, val);
+}
+
+// we piggyback on the existing sort criteria serialization
```

Comment:
The existing sort already provides a serialization/deserialization mechanism for sort types. I piggyback on this mechanism by converting commerce sort types to core sort ones, for deserialization and serialization.

Reply:
I've opted to write our own serialization/deserialization logic since it leads to simpler code